docs/source/tutorials/writing.md

```diff
@@ -66,6 +66,49 @@ Write array and tabular data.
 <DataFrameClient ['x', 'y']>
 ```
 
+In some scenarios, you may want to send data a chunk at a time, rather than sending the entire chunk at once. This might be in cases where the full data is not available at once, or the data is too large for memory. This can be achieved in two ways:
+
+The first one is to stack them before saving back to client using the above mentioned `write_array` method. This works when the size of data is small.
+
+When the size of merged data becomes an issue for memory, or in cases when you want to save the result on-the-fly as each individual array is generated, this could be achieved by using the `write_block` method with a pre-allocated space in client.
+
+```python
+# This approach will require you to know the final array dimension beforehand.
+
+# Assuming you have five 2d arrays (eg. images), each in shape of 32 by 32.
+>>> stacked_array_shape = (5, 32, 32)
+
+# Define a tiled ArrayStructure based on shape
+>>> import numpy
+>>> from tiled.structures.array import ArrayStructure
+
+>>> structure = ArrayStructure.from_array(numpy.zeros(stacked_array_shape, dtype = numpy.int8)) # A good practice to keep the dtype the same as your final results to avoid mismatch.
+>>> structure
+ArrayStructure(data_type=BuiltinDtype(endianness='not_applicable', kind=<Kind.integer: 'i'>, itemsize=1), chunks=((5,), (32,), (32,)), shape=(5, 32, 32), dims=None, resizable=False)
+
+# Re-define the chunk size to allow single array to be saved.
+>>> structure.chunks = ((1,) * stacked_array_shape[0], (stacked_array_shape[1],), (stacked_array_shape[2],))
+
+# Now to see that the chunk for the first axis has been divided.
+>>> structure
+ArrayStructure(data_type=BuiltinDtype(endianness='not_applicable', kind=<Kind.integer: 'i'>, itemsize=1), chunks=((1, 1, 1, 1, 1), (32,), (32,)), shape=(5, 32, 32), dims=None, resizable=False)
+
+# Allocate a new array client in tiled
+>>> array_client = client.new(structure_family="array", structure=structure, key ="stacked_result", metadata={"color": "yellow", "barcode": 13})
```

Comment:
Thanks for the suggestion. To answer the first part: I am using `tiled == 0.1.0a113` at this moment, and it worked out fine on my local end.

I will test the suggested `DataSource` and see if it's compatible with my pipeline. Will follow up on a new comment.

Reply:
Follow up: I went through the release notes, the `DataSource` is introduced in version `0.1.0a115`, so within my development environment (0.1.0a113) I was not able to run the suggested code.

To accommodate this, I will add both approaches and comment with notes about tiled version. This could serve as a temporary solution until a more advanced high level version come into the play.

Reply:
I think we should write the tutorial to the most recent version of tiled, given that it is still in alpha.